api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMediaDaoImpl.java

```diff
@@ -0,0 +1,144 @@
+	protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams) {
+		theParams.getParameters().forEach(entry -> {
+			switch (entry.getKey()) {
+				case FhirConstants.MEDIA_STATUS:
+					entry.getValue().forEach(status -> handleStatus(criteria, (TokenAndListParam) status.getParam()));
+					break;
+				case FhirConstants.MEDIA_TYPE:
+					entry.getValue().forEach(type -> handleMediaType(criteria, (TokenAndListParam) type.getParam()));
+					break;
+				case FhirConstants.MEDIA_SUBJECT:
+					entry.getValue().forEach(subject -> handleMediaSubject(criteria, (ReferenceAndListParam) subject.getParam()));
+					break;
+				case FhirConstants.MEDIA_ENCOUNTER_REFERENCE:
+					entry.getValue().forEach(encounter -> handleMediaEncounterReference(criteria, (ReferenceAndListParam) encounter.getParam()));
+					break;
+				case FhirConstants.MEDIA_CREATED_DATE_TIME:
+					entry.getValue().forEach(createdDateTime -> handleDateRange(createdDateTime.getPropertyName(),
+							(DateRangeParam) createdDateTime.getParam()).ifPresent(criteria::add));
+					break;
+				case FhirConstants.MEDIA_CONTENT_TYPE:
+					entry.getValue().forEach(contentType -> handleMediaContentType(criteria, (TokenAndListParam) contentType.getParam()));
+					break;
+				case FhirConstants.CONTENT_DATA:
+					entry.getValue().forEach(contentData -> handleContentData(criteria, (StringAndListParam) contentData.getParam()));
+					break;
+				case FhirConstants.CONTENT_TITLE:
+					entry.getValue().forEach(contentTitle -> handleContentTitle(criteria, (StringAndListParam) contentTitle.getParam()));
+					break;
+				case FhirConstants.CONTENT_DATE_OF_CREATION:
+					entry.getValue().forEach(contentDateOfCreation -> handleContentDateOfCreation(criteria, (DateRangeParam) contentDateOfCreation.getParam()));
+					break;
+				
+			}
+		});
+	}
+
+	private void handleStatus(Criteria criteria, TokenAndListParam status) {
+		if(status != null){
+			if(lacksAlias(criteria, "st")){
+				criteria.createAlias("status", "st");
+			}
+			convertStringStatusToBoolean(status);
+		}
+	}
+
+	private void handleMediaType(Criteria criteria, TokenAndListParam mediaType) {
+		if(mediaType != null){
+			if(lacksAlias(criteria, "mt")){
+				criteria.createAlias("mediaType", "mt");
+			}
+		}
+	}
+
+	private void handleMediaSubject(Criteria criteria, ReferenceAndListParam mediaSubject) {
+		if(mediaSubject != null){
+			if(lacksAlias(criteria, "ms")){
+				criteria.createAlias("mediaSubject", "ms");
+			}
+		}
+	}
+
+	private void handleMediaEncounterReference(Criteria criteria, ReferenceAndListParam encounterReference) {
+		if(encounterReference != null){
+			if(lacksAlias(criteria, "er")){
+				criteria.createAlias("mediaEncounterReference", "er");
+			}
+		}
+	}
+
+	private void handleMediaCreatedDate(Criteria criteria, DateRangeParam mediaCreatedDate) {
+		if(mediaCreatedDate != null){
+			if(lacksAlias(criteria, "dt")){
+					criteria.createAlias("mediaCreatedDate", "dt");
+			}
+		}
+	}
```

Comment:
This is superfluous. Ideally, you should re-use already existing functionality. i.e instead of this 👇🏾
```java
case FhirConstants.MEDIA_CREATED_DATE_TIME:
     entry.getValue().forEach(createdDateTime -> handleDateRange(createdDateTime.getPropertyName(),
	(DateRangeParam) createdDateTime.getParam()).ifPresent(criteria::add));
    break;
```
re-use `handleDateRange()` which somehow you are using it. Probably, you should replace with;
```java
case FhirConstants.DATE_RANGE_SEARCH_HANDLER:
    entry.getValue().forEach(dateRangeParam -> handleDateRange(dateRangeParam.getPropertyName(),
	(DateRangeParam) dateRangeParam.getParam()).ifPresent(criteria::add));
    break;
```
No need for the new constant and new method `handleMediaCreatedDate()`. This applies to `encounterReference` and the `subjectReference` - patient reference. Try re-using already existing functionality as much as possible.